WordPress/src/main/java/org/wordpress/android/inappupdate/InAppUpdateManagerNoop.kt

```diff
@@ -0,0 +1,21 @@
+package org.wordpress.android.inappupdate
+
+import android.app.Activity
+
+class InAppUpdateManagerNoop: IInAppUpdateManager {
```

Comment:
Hey @AjeshRPai,

Thanks for your feedback! I chose to use the Strategy Pattern with Interfaces because it helps keep the feature enablement decision out of the InAppUpdateManagerImpl.

I get your point about the simplicity and reduced overhead of checking the Feature Flag directly in InAppUpdateManagerImpl. It’s definitely a simpler way to go. However, I think using the strategy pattern keeps things cleaner and easier to manage.

I’m leaning towards sticking with this method for now, unless you think it might cause problems down the line. What’s your take?

Reply:
No strong opinions. Good to go with the strategy pattern